### PR TITLE
User login

### PR DESCRIPTION
**Pull request by acbk2b**

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -1,7 +1,90 @@
...
+import os
+import hashlib
 import sqlite3
+from flask_sqlalchemy import SQLAlchemy
+from flask_login import LoginManager, UserMixin, login_user, logout_user, current_user
+from flask import Flask, render_template, request, redirect, url_for
 app = Flask(__name__)
 
+# Secret key of random bytes (used for sessions and user login implementation)
+app.secret_key = b'\x15\xbe\x9bW\x80\xa1\x8d\xe0\x907\x08D\xfbr\x81\xf7'
```
Comment (cathyrae903) [inline server.py]: Do we want a secret key directly in the code?

File: server.py [hunk 2]
```diff
@@ -1,7 +1,90 @@
...
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(30), unique=True)
+    password = db.Column(db.String(32))
+    salt = db.Column(db.Integer)
+
+
+@app.route("/new_user", methods=["GET", "POST"])
+def new_user():
+    if request.method == "POST":
+        print("HELLO")
```
Comment (cathyrae903) [inline server.py]: Remove debug code

File: server.py (hunk 1 above)
Comment (cathyrae903) [reply server.py]: @acbk2b @GRNCheetah

File: server.py [hunk 3]
```diff
@@ -1,7 +1,97 @@
...
+app.config['SECRET_KEY'] = 'secretkey'
+db = SQLAlchemy(app)
+
+# Initialize Login Manager
+login_manager = LoginManager()
+login_manager.init_app(app)
+
+
+# Potential TODO: Get the submissions stuff set up on SQL alchemy for
+# consistency and easy of use throughout the program
```
Comment (cathyrae903) [inline server.py:33]: `ease of use`? 
Do we have time to do this? @acbk2b

File: server.py [hunk 4]
```diff
@@ -1,7 +1,97 @@
...
+    2 Attributes, an id (primary key) and a username
+    '''
+
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(30), unique=True)
+    password = db.Column(db.String(32))
+    salt = db.Column(db.Integer)
+
+
+@app.route("/new_user", methods=["GET", "POST"])
```
Comment (GRNCheetah) [inline server.py:46]: Was there a reason we accept GET requests?

Comment (GRNCheetah) [reply server.py:46]: nm i get it

File: server.py (hunk 1 above)
Comment (GRNCheetah) [reply server.py]: Make it random and I think this is good

**Review by cathyrae903 — APPROVED** [review]

**Review by GRNCheetah — APPROVED** [review]

**Pull request merged**